sorc/machine-setup.sh

```diff
@@ -67,16 +67,22 @@ elif [[ -d /data/prod ]] ; then
     fi
     target=s4
     module purge
-elif [[ "$(dnsdomainname)" =~ "pw" ]]; then
-    if [[ "${PW_CSP}" == "aws" ]]; then # TODO: Add other CSPs here.
-	target=noaacloud
-        module purge
+else
+    if [[ ! -v PW_CSP ]]; then
+        echo WARNING: UNSUPPORTED CSP PLATFORM 1>&2
+        echo WARNING: UNKNOWN PLATFORM 1>&2; exit 99
+    elif [[ -z "${PW_CSP}" ]]; then
+        echo WARNING: UNSUPPORTED CSP PLATFORM 1>&2
+        echo WARNING: UNKNOWN PLATFORM 1>&2; exit 99
     else
-        echo WARNING: UNSUPPORTED CSP PLATFORM 1>&2; exit 99
+        if [[ "${PW_CSP}" == "aws" || "${PW_CSP}" == "azure" || "${PW_CSP}" == "google" ]]; then
+            target=noaacloud
+            module purge
```

Comment:
On Hera, I tried to simulate an error on a non-supported, non-cloud platform:

```
 elif [[ -d /lfs/h1 ]] ; then
     target=wcoss2
     module reset
-elif [[ -d /scratch1 ]] ; then
+elif [[ -d /sratch1 ]] ; then
     # We are on NOAA Hera
```

I got two warnings - one for a CSP platform. 

```
++ echo WARNING: UNSUPPORTED CSP PLATFORM
WARNING: UNSUPPORTED CSP PLATFORM
++ echo WARNING: UNKNOWN PLATFORM
WARNING: UNKNOWN PLATFORM
```